opnode/bss/config.go

```diff
@@ -0,0 +1,224 @@
+	// lead to the two using the same wallet.
+	ErrSameSequencerAndL2OutputPrivKey = errors.New("sequencer-priv-key and " +
+		"l2-output-priv-key must be distinct")
+)
+
+type Config struct {
+	/* Required Params */
+
+	// BuildEnv identifies the environment this binary is intended for, i.e.
+	// production, development, etc.
+	BuildEnv string
+
+	// EthNetworkName identifies the intended Ethereum network.
+	EthNetworkName string
+
+	// L1EthRpc is the HTTP provider URL for L1.
+	L1EthRpc string
+
+	// L2EthRpc is the HTTP provider URL for L1.
+	L2EthRpc string
+
+	// CTCAddress is the CTC contract address.
+	CTCAddress string
+
+	// SROAddress is the SCC contract address.
+	SROAddress string
+
+	// MinL1TxSize is the minimum size in bytes of any L1 transactions generated
+	// by the batch submitter.
+	MinL1TxSize uint64
+
+	// MaxL1TxSize is the maximum size in bytes of any L1 transactions generated
+	// by the batch submitter.
+	MaxL1TxSize uint64
+
+	// MaxTxBatchCount is the maximum number of L2 transactions that can ever be
+	// in a batch.
+	MaxTxBatchCount uint64
+
+	// MaxStateBatchCount is the maximum number of L2 state roots that can ever
+	// be in a batch.
+	MaxStateBatchCount uint64
+
+	// MaxBatchSubmissionTime is the maximum amount of time that we will
+	// wait before submitting an under-sized batch.
+	MaxBatchSubmissionTime time.Duration
+
+	// PollInterval is the delay between querying L2 for more transaction
+	// and creating a new batch.
+	PollInterval time.Duration
+
+	// NumConfirmations is the number of confirmations which we will wait after
+	// appending new batches.
+	NumConfirmations uint64
+
+	// ResubmissionTimeout is time we will wait before resubmitting a
+	// transaction.
+	ResubmissionTimeout time.Duration
+
+	// FinalityConfirmations is the number of confirmations that we should wait
+	// before submitting state roots for CTC elements.
+	FinalityConfirmations uint64
+
+	// RunTxBatchSubmitter determines whether or not to run the tx batch
+	// submitter.
+	RunTxBatchSubmitter bool
+
+	// RunStateBatchSubmitter determines whether or not to run the state batch
+	// submitter.
+	RunStateBatchSubmitter bool
+
+	//SafeMinimumEtherBalance is the safe minimum amount of ether the batch
+	//submitter key should hold before it starts to log errors.
+	SafeMinimumEtherBalance uint64
+
+	// ClearPendingTxs is a boolean to clear the pending transactions in the
+	// mempool on startup.
+	ClearPendingTxs bool
+
+	/* Optional Params */
+
+	// LogLevel is the lowest log level that will be output.
+	LogLevel string
+
+	// BlockOffset is the offset between the CTC contract start and the L2 geth
+	// blocks.
+	BlockOffset uint64
+
+	// SequencerPrivateKey the private key of the wallet used to submit
+	// transactions to the CTC contract.
+	SequencerPrivateKey string
+
+	// PropopserPrivateKey the private key of the wallet used to submit
+	// transaction to the SCC contract.
+	L2OutputPrivateKey string
```

Comment:
These config options were used at one point with the old batch submitter

Reply:
yes this config option is used in various test environments in the monorepo. i went ahead and removed the option to set private keys explicitly since this test environment already uses mnemonics